libsolidity/parsing/Token.cpp

```diff
@@ -151,20 +153,19 @@ tuple<Token::Value, unsigned int, unsigned int> Token::fromIdentifierOrKeyword(s
 				positionM < positionX &&
 				positionX < _literal.end() &&
 				*positionX == 'x' &&
-				all_of(positionX + 1, _literal.end(), ::isdigit)
+				all_of(++positionX, _literal.end(), ::isdigit)
```

Comment:
The changes for m and n needed to be put there, and the issue with UFixedMxN not being considered an identifier are also solved below.